pkg/gofr/datasource/file/fs.go

```diff
@@ -61,3 +61,26 @@ func (fileSystem) RemoveAll(path string) error {
 func (fileSystem) Rename(oldname, newname string) error {
 	return os.Rename(oldname, newname)
 }
+
+func (fileSystem) Stat(name string) (FileInfo, error) {
+	return os.Stat(name)
+}
+
+func (fileSystem) CurrentDir() (string, error) {
+	return os.Getwd()
+}
+
+func (fileSystem) ChangeDir(dir string) error {
+	return os.Chdir(dir)
+}
+
+func (fileSystem) ReadDir(dir string) ([]FileInfo, error) {
+	entries, err := os.ReadDir(dir)
+
+	fileInfo := make([]FileInfo, len(entries))
+	for i := range entries {
+		fileInfo[i], _ = entries[i].Info()
```

Comment:
I wouldn't ignore the error here

Reply:
I assumed that if there is any error getting the file info... then fileInfo[i] would be nil anyway....But I guess calling methods on nil can cause panic .....So I will add continue statement on error ..so that it ignores the file that gave error and appends the rest. 

Does this sound reasonable or should I just return ?

Reply:
I would say return an error on first error.

It may never happen, but at least it would be covered